### DISCUSSION TITLE
How to intelligently add local sizes in pyfluent and conveniently obtain zone name

### DISCUSSION THREAD

**Discussion by Mr-HB833** [ansys/pyfluent#1922] [Q&A]
I now want to use pyfluent to add local dimensions in fluent meshing, as shown in the figure. I would like to select zone name by label as "oil_fluid-oil_fluid_0", "oil_fluid-oil_fluid_1"...'oil_fluid oil_fluid_5'(six zone). I successfully created a local size using the following code, but now I want to make my code more intelligent because the number of zones I have may change in the future. I would like to know if there is a way to directly select all zones with 'oil_fluid oil_fluid' in their names. Regardless of whether the number of zones increases or decreases, I can quickly select the zone I want. What can i do? Please help me, thank you very much!!!!
![image](https://github.com/ansys/pyfluent/assets/110915109/8fff93b1-a9b9-4592-816b-f122700e774e)
`meshing_session.workflow.TaskObject["Add Local Sizing"].Arguments.setState(AddChild="yes",
                                                                               BOIControlName="bodysize_oil",
                                                                               BOIExecutio="Body Size",
                                                                               BOIFaceLabelList=["oil_fluid-oil_fluid_0",
                                                                                                 "oil_fluid-oil_fluid_1",
                                                                                                 "oil_fluid-oil_fluid_2",
                                                                                                 "oil_fluid-oil_fluid_3",
                                                                                                 "oil_fluid-oil_fluid_4",
                                                                                                 "oil_fluid-oil_fluid_5"],
                                                                               BOISize=2)`

**Comment by Mr-HB833** [comment]
i got it. I used the following code to select all zones with "oil_fluid oil_fluid" in their names. At present, it has solved my problem, but I also want to know if there are other ways to obtain zone name by label to achieve intelligent and convenient selection of zone name
`meshing_session.workflow.TaskObject["Add Local Sizing"].Arguments.setState(
    AddChild="yes",
    BOIControlName="bodysize_oil",
    BOIExecutio="Body Size",
    BOIFaceLabelList='*oil_fluid-oil_fluid*',
    BOISize=2)`

**Comment by Mr-HB833** [comment]
Now I have new problem about the failure of wildcards in the solver. when I use this code:`solver_session.setup.boundary_conditions.pressure_inlet['b_ inlet'].p0()`
I can get the following results:`{'option': 'constant or expression', 'constant':0}` 
But I used wildcards to run the code:` solver_session.setup.boundary_conditions.pressure_inlet['*b_inlet*'].p0()`
The following error in picture.
![image](https://github.com/ansys/pyfluent/assets/110915109/c0ffc81e-63fa-48fb-8002-5f6c7ea295e9)
![image](https://github.com/ansys/pyfluent/assets/110915109/672fc4a4-5eb7-430e-9d26-24ab870ddfb6)
![image](https://github.com/ansys/pyfluent/assets/110915109/1b1a4b4f-575d-402b-87c2-b62a6f3e80b0)

**Reply by mkundu1** [reply]
Can you please share the Fluent and PyFluent versions?
